src/modules/universal-discovery/components/finder/finder.component.js

```diff
@@ -18,39 +19,142 @@ export default class FinderComponent extends Component {
-        this.props.findLocationsByParentLocationId(
-            Object.assign({}, this.props.restInfo, { parentLocationId: this.props.startingLocationId}),
-            this.updateLocationsData
-        );
+        const isForcedLocation = this.props.allowedLocations.length === 1;
+        const allowedIncludesPreselected = this.props.allowedLocations.includes(this.props.preselectedLocation);
+        const isPreselectedLocation = this.props.preselectedLocation && (!this.props.allowedLocations.length || allowedIncludesPreselected);
+
+        if (isForcedLocation) {
+            this.loadPreselectedData(this.props.allowedLocations[0], true);
+        } else if (isPreselectedLocation) {
+            this.loadPreselectedData(this.props.preselectedLocation, true);
+        } else {
+            this.props.findLocationsByParentLocationId(
+                Object.assign({}, this.props.restInfo, { parentLocationId: this.props.startingLocationId }),
+                this.updateLocationsData
+            );
+        }
     }
 
     componentDidUpdate() {
         this.updateBranchesContainerScroll();
     }
 
+    componentWillReceiveProps(props) {
+        if (this.preselectedItem && this.locationsMap && this.activeLocations && props.isVisible && props.isVisible !== this.props.isVisible) {
+            this.setState(state => Object.assign({}, state, {
+                locationsMap: this.locationsMap,
+                activeLocations: this.activeLocations
+            }), () => this.props.isVisible && this.props.onItemSelect(this.preselectedItem));
+        }
+    }
+
+    /**
+     * Load data of preselected location.
+     *
+     * @method loadPreselectedData
+     * @param {String} locationId
+     * @memberof FinderComponent
+     */
+    loadPreselectedData(id) {
+        const promise = new Promise(resolve => loadAccordionData(
+            {startingLocationId: this.props.startingLocationId, locationId: id},
+            resolve
+        ));
+
+        promise
+            .then((response) => {
+                this.createAccordionData(response);
+                this.preselectedItem = response.locations[id].Location;
+                this.setAccordionData();
+            });
+    }
+
+    /**
+     * Create an accordion data.
+     *
+     * @method createAccordionData
+     * @param {Object} locations
+     * @param {Object} subitems
+     * @memberof FinderComponent
+     */
+    createAccordionData({locations, subitems}) {
+        const createItem = ({parent, offset = 0, children, location}) => {
+            const createData = (items) => {
+                return items.locations.map(location => {
+                    return {value: location};
+                });
+            };
+            const data = children ? createData(children) : [];
+            const count = children ? children.totalCount : location.childCount;
+
+            return {count, data, offset, parent};
+        };
+
+        if (subitems[1]) {
```

Comment:
Looks strange. If a second subitem exists, then you're creating eveything manually. I need some explanation here.

Reply:
Subitems is an object with keys as locationId. I'm checking if in subitems there is a location with id `1` (This is a special case because there is no such thing like location with id 1 but you can load children of this "location").